crates/shared/src/sources/balancer_v2/pool_fetching/pool_storage.rs

```diff
@@ -174,23 +174,6 @@ where
             .max()
             .unwrap_or_default()
     }
-}
-
-#[async_trait::async_trait]
-impl<Factory> EventStoring<BasePoolFactoryEvent> for PoolStorage<Factory>
-where
-    Factory: FactoryIndexing,
-{
-    async fn replace_events(
-        &mut self,
-        events: Vec<Event<BasePoolFactoryEvent>>,
-        range: RangeInclusive<BlockNumber>,
-    ) -> Result<()> {
-        tracing::debug!("replacing {} events for block {:?}", events.len(), range);
-
-        self.remove_pools_newer_than_block(range.start().to_u64());
-        self.append_events(events).await
-    }
 
     async fn append_events(&mut self, events: Vec<Event<BasePoolFactoryEvent>>) -> Result<()> {
```

Comment:
No new code, just `append_events` is moved out of `impl<Factory> EventStoring<BasePoolFactoryEvent> for PoolStorage<Factory>`